src/main/java/ru/mts/bpp/CreateAnimalServiceBeanPostProcessor.java

```diff
@@ -0,0 +1,41 @@
+package ru.mts.bpp;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Configuration;
+import ru.mts.animalsCreators.AnimalFactory;
+import ru.mts.animalsCreators.CreateAnimalServiceLmpl;
+
+import java.util.Random;
+
+@Configuration
+public class CreateAnimalServiceBeanPostProcessor implements BeanPostProcessor {
+    /**
+     * <b>postProcessBeforeInitialization</b>
+     * @param bean the new bean instance
+     * @param beanName the name of the bean
+     * Если созданный бин это createanimalservicelmpl, то заполняет поле animalType случайным типом
+     * @return bean
+     * @throws BeansException
+     */
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        if (beanName.equalsIgnoreCase("createanimalservicelmpl")) {
```

Comment:
```suggestion
        if (beanName.equalsIgnoreCase("createanimalserviceimpl")) {
```
не забудь после того как исправишь имя класса